crates/solidity/testing/perf/npm/src/benchmarks/subjects/solidity.parser.subject.mts

```diff
@@ -0,0 +1,42 @@
+import { parse } from "@solidity-parser/parser";
+import { log, Subject, SolidityProject, Timings } from "../common.mjs";
+
+export class AntlrSubject implements Subject {
+  public name = "solidity parser";
+
+  async test(project: SolidityProject, file: string): Promise<Timings> {
+    const start = performance.now();
+
+    let toProcess = new Array<string>(file);
+    let processed = new Set<string>();
+
+    log("Start");
+
+    while (toProcess.length > 0) {
+      const filePath = toProcess.pop()!;
+
+      if (processed.has(filePath)) {
+        continue;
+      }
+
+      log(`To process ${filePath}`);
+      processed.add(filePath);
+
+      const content = project.fileContents(filePath);
+      const result = parse(content, { tolerant: true, loc: true });
+      if (result.errors) {
+        console.error("Errors during parsing with solidity-parser:");
+        console.error(result.errors);
+        process.exit(-1);
+      }
+
+      const imports = result.children.filter((f) => f.type == "ImportDirective");
+
+      imports.forEach((imprt) => {
+        toProcess.push(project.resolveImport(filePath, imprt.path));
```

Comment:
My understanding is that `solc` parses/binds all input files, regardless of imports, right?

In ANTLR, I wonder what is the reason we are resolving/traversing the import tree, rather than simply parsing all of them?
Same question for Slang: should we be calling `addFile()` on all files for this reason?

Reply:
no, solc is following imports just like us. If you add a file with garbage in the .json file, it's ignored.